mlx_graphs/utils/scatter.py

```diff
@@ -0,0 +1,46 @@
+from typing import Union, List
+
+import mlx.core as mx
+
+from mlx_graphs.utils.utils import max_nodes
+
+scattering_operations = ["add", "sum", "max"]
+
+
+def scatter(values: mx.array, indices: mx.array, out_shape: Union[List[int], int], aggr: str, axis: int = 0):
+    if aggr not in scattering_operations:
+        raise NotImplementedError(f"Aggregation {aggr} not implemented yet.")
+
+    empty_tensor = mx.zeros(out_shape, dtype=values.dtype)
+
+    update_dim = (values.shape[0], 1, *values.shape[1:])
+    values = values.reshape(update_dim)
+
+    if aggr in ["add", "sum"]:
+        return scatter_add(empty_tensor, indices, values)
+    if aggr == "max":
+        return scatter_max(empty_tensor, indices, values)
+
+
+def scatter_add(src: mx.array, indices: mx.array, values: mx.array, axis: int = 0):
+    return mx.scatter_add(src, indices, values, axis)
+
+
+def scatter_max(src: mx.array, indices: mx.array, values: mx.array, axis: int = 0):
+    return mx.scatter_max(src, indices, values, axis)
```

Comment:
I'm not sure where we stand on this in `mlx`. Are we back to having the `scatter_{ops}` implemented? I understood we were going with the `.at[]` syntax.

Reply:
You're right, I will not let these, just using them locally in my mlx code that's why. Will remove them once scatter_max vjp is merged in mlx